trade-finance-manager-ui/server/controllers/case/activity/index.test.js

```diff
@@ -71,3 +78,212 @@ describe('GET activity', () => {
+  });
+
+  it('should render filtered activities with data', async () => {
+    const req = {
+      params: {
+        _id: mockDeal._id,
+      },
+      body: {
+        filterType: CONSTANTS.ACTIVITIES.ACTIVITY_FILTERS.COMMENT,
+      },
+      session,
+    };
+
+    const activities = activityController.mappedActivities(mockDeal.tfm.activities);
+
+    await activityController.filterActivities(req, res);
+    expect(res.render).toHaveBeenCalledWith('case/activity/activity.njk',
+      {
+        activePrimaryNavigation: 'manage work',
+        activeSubNavigation: 'activity',
+        deal: mockDeal.dealSnapshot,
+        tfm: mockDeal.tfm,
+        dealId: mockDeal.dealSnapshot._id, // eslint-disable-line no-underscore-dangle
+        user: session.user,
+        selectedActivityFilter: CONSTANTS.ACTIVITIES.ACTIVITY_FILTERS.COMMENT,
+        activities,
+      });
+  });
+});
+
+describe('getCommentBox()', () => {
+  const mockDeal = {
+    _id: '1000023',
+    dealSnapshot: {
+      _id: '1000023',
+      submissionDetails: {
+        supplierName: 'test supplier',
+      },
+    },
+    tfm: {
+      activities: [],
+    },
+  };
+
+  beforeEach(() => {
+    api.getDeal = () => Promise.resolve(mockDeal);
+  });
+
+  it('should render the comment box', async () => {
+    const req = {
+      params: {
+        _id: mockDeal._id,
+      },
+      session,
+    };
+
+    await activityController.getCommentBox(req, res);
+    expect(res.render).toHaveBeenCalledWith('case/activity/activity-comment.njk',
+      {
+        dealId: mockDeal.dealSnapshot._id,
+        user: session.user,
+        maxCommentLength: 1000,
+      });
+  });
+});
+
+describe('postComment()', () => {
+  const mockDeal = {
+    _id: '1000023',
+    dealSnapshot: {
+      _id: '1000023',
+      submissionDetails: {
+        supplierName: 'test supplier',
+      },
+    },
+    tfm: {
+      activities: [],
+    },
+  };
+
+  const longComment = 'Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Aenean commodo ligula eget dolor. Aenean massa. Cum sociis natoque penatibus et magnis dis parturient montes, nascetur ridiculus mus. Donec quam felis, ultricies nec, pellentesque eu, pretium quis, sem. Nulla consequat massa quis enim. Donec pede justo, fringilla vel, aliquet nec, vulputate eget, arcu. In enim justo, rhoncus ut, imperdiet a, venenatis vitae, justo. Nullam dictum felis eu pede mollis pretium. Integer tincidunt. Cras dapibus. Vivamus elementum semper nisi. Aenean vulputate eleifend tellus. Aenean leo ligula, porttitor eu, consequat vitae, eleifend ac, enim. Aliquam lorem ante, dapibus in, viverra quis, feugiat a, tellus. Phasellus viverra nulla ut metus varius laoreet. Quisque rutrum. Aenean imperdiet. Etiam ultricies nisi vel augue. Curabitur ullamcorper ultricies nisi. Nam eget dui. Etiam rhoncus. Maecenas tempus, tellus eget condimentum rhoncus, sem quam semper libero, sit amet adipiscing sem neque sed ipsum. Na';
+  const normalComment = 'test';
```

Comment:
`a.repeat(1001)`

Reply:
this may take a while as it needs to write each character one by one. as an alternative, we could do `'aaaaa'.repeat(201)`. so it repeats that block (in theory?), 201 times